src/bls12_381/fq.rs

```diff
@@ -552,37 +551,15 @@ impl BaseFromRO for Fq {
 }
 
 impl Signum0 for Fq {
-    // returns the sign of a center lifted element over the integer ring
     fn sgn0(&self) -> Sgn0Result {
-        // (group_order - 1)/2
-        const PM1DIV2: FqRepr = FqRepr([
-            0xdcff7fffffffd555u64,
-            0x0f55ffff58a9ffffu64,
-            0xb39869507b587b12u64,
-            0xb23ba5c279c2895fu64,
-            0x258dd3db21a5d66bu64,
-            0x0d0088f51cbff34du64,
-        ]);
-
-        if self.into_repr().cmp(&PM1DIV2) == Ordering::Greater {
+        if self.into_repr().0[0] & 1 == 1 {
```

Comment:
Are you sure the array is always of length at least 1 ? And is there no way to use the same style of equality as before ?

Reply:
Yes `fq`s representation is a fixed size array of 6 elements. 
This is the check the spec recommends, so I am using that (x mod 2)